helix-term/src/ui/statusline.rs

```diff
@@ -487,6 +487,16 @@ where
         .version_control_head()
         .unwrap_or_default()
         .to_string();
+    if head.is_empty() {
+        return;
+    }
 
-    write(context, head, None);
+    let config = context.editor.config.load();
+    let version_control_icon = &config.statusline.version_control_icon;
+    let text = match version_control_icon.is_empty() {
+        false => format!(" {} {head} ", version_control_icon),
+        true => format!(" {head} "),
```

Comment:
```suggestion
        false => format!("{} {head}", version_control_icon),
        true => format!("{head}"),
```

spaces should never be included inside statusline elements directly we have the spacer element for that

Reply:
The reason for having those spaces there was that I thought the element looked bad without any padding if you added a background colour for it.

Padding:
![image](https://user-images.githubusercontent.com/31644842/231953527-fa09ad35-f5af-4cf7-b505-279661d5ef7d.png)
No padding:
![image](https://user-images.githubusercontent.com/31644842/231954428-58e9417c-8003-4f21-9e70-8eeb4363bab0.png)